packages/material-ui/src/Popper/Popper.test.js

```diff
@@ -169,6 +167,32 @@ describe('<Popper />', () => {
     });
   });
 
+  describe('prop: keepMounted', () => {
+    it('should keep the children in the DOM', () => {
+      const wrapper = mount(<Popper {...defaultProps} open={false} keepMounted />);
+      assert.strictEqual(wrapper.find('span').length, 1);
+    });
+
+    /* Test case for https://github.com/mui-org/material-ui/issues/15180 */
+    it('should remove from DOM when closed whilst transition has entering status', () => {
+      const wrapper = mount(
+        <Popper {...defaultProps} open={false} keepMounted={false} transition>
+          {({ TransitionProps }) => (
+            <Grow {...TransitionProps}>
+              <span>Hello World</span>
+            </Grow>
+          )}
+        </Popper>,
+      );
+      wrapper.setProps({ open: true });
+      wrapper.update();
+      assert.strictEqual(wrapper.find('span').length, 1);
+      assert.strictEqual(wrapper.find('Transition').instance().state.status, 'entering');
+      wrapper.setProps({ open: false });
+      assert.strictEqual(wrapper.find('span').length, 0);
```

Comment:
That's working differently than the first setProps. When we setProps({ open: true }) we need the wrapper.update call but here we don't.

Reply:
Interestingly, changing from using a `<Grow>` component to a `<Transition>` component and applying the props needed to replicate the `next` failure now means I need a `wrapper.update()` after the `wrapper.setProps({ open: false })`. That at least aligns it with the other tests, but still not truly understanding why it's required at all. [Looks like there's a lot of longstanding confusion about `.update`, though](https://github.com/airbnb/enzyme/issues/1153)